libs/saveFiles.js

```diff
@@ -38,22 +38,29 @@ const sanitizeEntry = function (entry) {
   return entry
 }
 
-const downloadEntry = function (entry, folderPath) {
-  const reqOptions = Object.assign(
-    {
-      uri: entry.fileurl,
-      method: 'GET',
-      jar: true
-    },
-    entry.requestOptions
-  )
+const downloadEntry = function (entry, options) {
+  const reqOptions = Object.assign({
+    uri: entry.fileurl,
+    method: 'GET',
+    jar: true
+  }, entry.requestOptions)
+
   return cozy.files
-    .statByPath(folderPath)
+    .statByPath(options.folderPath)
     .then(folder => {
-      return cozy.files.create(rq(reqOptions), {
+      let filePromise = rq(reqOptions)
+      const createFileOptions = {
         name: getFileName(entry),
-        dirID: folder._id
-      })
+        dirID: folder._id,
+        contentType: options.contentType
+      }
+      if (options.fileStreamProcess) {
```

Comment:
I think `options.postProcess` would be easier to understand for the user.

Reply:
Ah, `postProcess` already exists, why can't it be used here ?

Reply:
I agree with postProcessFile.

You are right saveEntry does too much